pyanaconda/kickstart_dispatcher/element.py

```diff
@@ -0,0 +1,239 @@
+# Containers for working with kickstart elements
+#
+# Copyright (C) 2017  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from enum import Enum
+
+
+class KickstartElement(object):
+    """Stores element parsed from kickstart with reference to file.
+
+    Element can be a command, a section or an addon. Addon is not considered
+    as a section. The type is infered from args argument.
+    """
+
+    class KickstartElementType(Enum):
+        COMMAND = "command"
+        SECTION = "section"
+        ADDON = "addon"
+
+    def __init__(self, args, lines, lineno, filename):
+        """Construct the element from data obtained by a KickstartParser.
+
+        :param args: tokens parsed from a line by shlex in KickstartParser.
+                     - for command the whole command
+                     - for section its header
+                     - for addon its header
+        :type args: list(str)
+        :param lines: lines parsed by KickstarParser
+                     - for command the whole command
+                     - for section its body
+                     - for addons its body
+        :type lines: list(str)
```

Comment:
I think `content` would be more appropriate than `lines` and save the original to something like `original_content` or better don't save it at all, what do you think?

Reply:
I think content would be misleading here: the problem is that kickstart parser gives us different data for commands and sections. For commands - line is the content, for sections - lines is only the body content (there is no header line supplied, only in form of args - shlex parsed line).  If we find we want to construct KickstartElements in other contexts than kickstart parsing, let's add support/api for it later. (Trying to be generic when creating element I ended up with something like v1 of the patch was (append_command, append_section, ...). But I think we should be careful with allowing to construct the Elements in other ways than as it is done from kickstart parsing. The KickstartElement evolved to a class having also the knowledge about kickstart parser parsing callbacks output, and even re-composing kickstart content from parser output (for sections). And I think it is OK as the purpose of KickstartElements is to be basically a read-once structured storage for kickstart.

Reply:
Ok, thanks for explanation.